pkg/registry/storage/volumeattachment/storage/storage.go

```diff
@@ -46,5 +56,33 @@ func NewREST(optsGetter generic.RESTOptionsGetter) *REST {
 		panic(err) // TODO: Propagate error up
 	}
 
-	return &REST{store}
+	statusStore := *store
+	statusStore.UpdateStrategy = volumeattachment.StatusStrategy
+
+	return &VolumeAttachmentStorage{
+		VolumeAttachment: &REST{store},
+		Status:           &StatusREST{store: &statusStore},
+	}
+}
+
+// StatusREST implements the REST endpoint for changing the status of a VolumeAttachment
+type StatusREST struct {
+	store *genericregistry.Store
+}
```

Comment:
follow-up: please add a type assertion to ensure this implements enough to patch status:

`var _ = rest.Patcher(&StatusREST{})`